common/src/main/java/org/geysermc/floodgate/link/SQLiteImpl.java

```diff
@@ -0,0 +1,97 @@
+        getLogger().info("Loading Floodgate linked player database...");
+        try {
+            Class.forName("org.sqlite.JDBC");
+
+            // create a database connection
+            connection = DriverManager.getConnection("jdbc:sqlite:" + databasePath.toString());
+            Statement statement = connection.createStatement();
+            statement.setQueryTimeout(30);  // set timeout to 30 sec.
+            statement.executeUpdate("create table if not exists LinkedPlayers (bedrockId string, javaUniqueId string, javaUsername string)");
+        } catch (ClassNotFoundException e) {
+            getLogger().severe("The required class to load the SQLite database wasn't found");
+        } catch (SQLException e) {
+            getLogger().log(Level.SEVERE, "Error while loading database", e);
+        }
+    }
+
+    @Override
+    public LinkedPlayer getLinkedPlayer(UUID bedrockId) {
+        // TODO: make it work with Java player UUIDs
+        try {
+            PreparedStatement query = connection.prepareStatement("select * from LinkedPlayers where bedrockId = ?");
+            query.setString(1, bedrockId.toString());
+            ResultSet result = query.executeQuery();
+            if (!result.next()) return null;
+
+            String javaUsername = result.getString("javaUsername");
+            UUID javaUniqueId = UUID.fromString(result.getString("javaUniqueId"));
+            return createLinkedPlayer(javaUsername, javaUniqueId, bedrockId);
+        } catch (SQLException | NullPointerException e) {
+            getLogger().log(Level.SEVERE, "Error while getting LinkedPlayer", e);
+            return null;
+        }
+    }
+
+    @Override
+    public boolean isLinkedPlayer(UUID bedrockId) {
+        try {
+            PreparedStatement query = connection.prepareStatement("select javaUniqueId from LinkedPlayers where bedrockId = ? or javaUniqueId = ?");
+            query.setString(1, bedrockId.toString());
+            query.setString(2, bedrockId.toString());
+            ResultSet result = query.executeQuery();
+            return result.next();
+        } catch (SQLException | NullPointerException e) {
+            getLogger().log(Level.SEVERE, "Error while checking if player is a LinkedPlayer", e);
+            return false;
+        }
+    }
+
+    @Override
+    public boolean linkPlayer(UUID bedrockId, UUID uuid, String username) {
+        try {
+            PreparedStatement query = connection.prepareStatement("insert into LinkedPlayers values(?, ?, ?)");
+            query.setString(1, bedrockId.toString());
+            query.setString(2, uuid.toString());
+            query.setString(3, username);
+            query.executeUpdate();
+            return true;
+        } catch (SQLException | NullPointerException e) {
+            getLogger().log(Level.SEVERE, "Error while linking player", e);
+            return false;
+        }
+    }
+
+    @Override
+    public boolean unlinkPlayer(UUID uuid) {
```

Comment:
Use a CompletableFuture here since direct database lookups can freeze the main thread if they take too long. This may require some changes in the `PlayerLink` class too.